lib/r10k/util/platform.rb

```diff
@@ -14,6 +16,16 @@ def self.platform
         end
       end
 
+      # We currently only suport FIPS mode on redhat 7, where it is
+      # toggled via a file.
+      def self.fips?
+        if File.exist?(FIPS_FILE)
+          File.read(FIPS_FILE).chomp == "1"
```

Comment:
I'm kinda surprised the r10k user would be able to read this file

Reply:
I'll be damned, it's world readable:
```
root@l3ieludaqizuhkz ~]# ls -la /proc/sys/crypto/
total 0
dr-xr-xr-x 1 root root 0 Jul 29 20:30 .
dr-xr-xr-x 1 root root 0 Jul 29 20:28 ..
-r--r--r-- 1 root root 0 Jul 29 20:30 fips_enabled
```

Reply:
Maybe it can't? Is the r10k user usually different than the puppet user? Because that's how puppet is doing it... see https://github.com/puppetlabs/puppet/blob/ba4d1a1aba0095d3c70b98fea5c67434a4876a61/lib/puppet/util/platform.rb#L24-L26. (This also worked in my manual testing, but there I was running as root on a pooler machine, which I guess might not be typical?)

Reply:
Works for me :P